Run pumount before umount so the hotplugging people will be happy.

http://bugs.gentoo.org/91977

--- eject.c
+++ eject.c
@@ -540,11 +540,14 @@
 	switch (fork()) {
 	  case 0: /* child */
 		  seteuid(getuid()); /* reduce likelyhood of security holes when running setuid */
-		  if(p_option)
-			  execl("/bin/umount", "/bin/umount", fullName, "-n", NULL);
-		  else
-			  execl("/bin/umount", "/bin/umount", fullName, NULL);
-		  fprintf(stderr, _("%s: unable to exec /bin/umount of `%s': %s\n"),
+		  if(p_option) {
+			  execlp("pumount", "pumount", fullName, "-n", NULL);
+			  execlp("umount", "umount", fullName, "-n", NULL);
+		  } else {
+			  execlp("pumount", "pumount", fullName, NULL);
+			  execlp("umount", "umount", fullName, NULL);
+		  }
+		  fprintf(stderr, _("%s: unable to exec umount of `%s': %s\n"),
 				  programName, fullName, strerror(errno));
 		  exit(1);
 		  break;
